sql/core/src/main/scala/org/apache/spark/sql/package.scala

```diff
@@ -111,6 +111,26 @@ package object sql {
     }
   }
 
+  private[sql] def withOrigin[T](
+      pysparkLoggingInfo: Option[java.util.Map[String, String]] = None)(f: => T): T = {
+    if (CurrentOrigin.get.stackTrace.isDefined) {
+      f
+    } else {
+      val st = Thread.currentThread().getStackTrace
```

Comment:
Can we keep the Java stacktrace as it is?

In PySpark we provide users an option `pysparkJVMStacktraceEnabled` to turn Java stacktrace on or off, since maybe someone still wants to see the Java stacktrace for deeper debugging.

Reply:
Maybe can we keep the Java stacktrace as it is?

In PySpark we also provide users an option pysparkJVMStacktraceEnabled to turn Java stacktrace on or off, since maybe someone still wants to see the Java stacktrace for deeper debugging?

WDYT, @HyukjinKwon ?

Reply:
I see

Reply:
Can we add a method to avoid duplicated code of capturing java stacktraces?

Reply:
> Can we add a method to avoid duplicated code of capturing java stacktraces?

Sure, will address it

Reply:
Remove duplication by adding a method